Sources/SwiftParserDiagnostics/DiagnosticExtensions.swift

```diff
@@ -126,10 +126,10 @@ extension FixIt.MultiNodeChange {
     var presentNode = MissingNodesBasicFormatter(viewMode: .fixedUp).visit(Syntax(node))
     presentNode = PresentMaker().rewrite(presentNode)
 
-    if let leadingTrivia {
+    if let leadingTrivia, !leadingTrivia.isEmpty {
```

Comment:
This doesn’t seem correct to me. I think you should be able to explicitly specify that you want empty trivia. That’s why we have the distinction between empty and `nil`. IMO, the solution here would be figure out what’s calling this and make it pass `nil` instead.

Reply:
Its happening here
https://github.com/apple/swift-syntax/blob/3ed9dcf27c69b991d3cd8eadd64764638f3b0dec/Sources/SwiftParserDiagnostics/ParseDiagnosticsGenerator.swift#L172-L173. 

We could change it to this `misplacedToken.leadingTrivia.isEmpty ? nil : misplacedToken.leadingTrivia` etc.

Reply:
Yes, I think that would make sense with a comment like the following

```swift
// Transfer any existing trivia. If there is no trivia in the misplaced token, pass `nil` so that `makePresent` can add required trivia, if necessary.
```